custom_components/plugwise/gateway.py

```diff
@@ -176,24 +177,27 @@ def __init__(self, coordinator, dev_id, name, model, vendor, fw):
         super().__init__(coordinator)
 
         self._attr_available = super().available
+        entry = coordinator.config_entry
         gw_id = coordinator.data[0]["gateway_id"]
-        self._attr_device_info = {
-            "identifiers": {(DOMAIN, dev_id)},
-            "name": name
+        self._attr_device_info = DeviceInfo(
+            configuration_url=f"http://{entry.data[CONF_HOST]}"
+            if dev_id == gw_id
+            else None,
```

Comment:
Let's avoid all this inline if/else stuff, it makes it really hard to read and easy to bug out on.

Reply:
Sure, I don't mind. Unfortunately I'm still pretty new to python, what kind of structure do you propose?

Reply:
determine first, use the variable later.